packages/server-wallet/docs/index.md

```diff
@@ -0,0 +1,183 @@
+# What is a state channel wallet?
+
+A node in a state channel network. See [nitro protocol](https://docs.statechannels.org/`).
+
+More concretely, a state channel wallet _is_:
+
+- a queryable store of keys, states and signatures
+- a run-loop, triggered by
+
+  - blockchain events
+  - API calls
+  - messages from counterparties
+
+  that
+
+  - signs states and transactions
+  - stores them
+  - passes them to external actors
+
+- an event emitter
+
+We usually talk of “The App” as the actor that makes the API calls and binds to the events. The wallet can be a library or run in a separate process to The App (e.g. on a server, or run in an iframe).
+
+A state channel wallet _is not_:
+
+- equipped with a graphical user interface (although this might be a desired plugin)
+
+A state channel wallet has the following _runtime dependencies_:
+
+- a message delivery service (deferred to The App)
+- a blockchain provider (e.g. infura)
+- a database (could be a Postgres server or indexedDB)
+
+The following diagram describes the various components in the architecture. The scope of this document is limited to the components left of the dividing line.
```

Comment:
How is this diagram generated? Without an easy way of generating it, it's not that maintainable. (Though, we shouldn't be updating this doc all that often.)

Reply:
I added a comment about where the diagram was created in this commit https://github.com/statechannels/statechannels/pull/3260/commits/dc92169b273699918abfc673906a0bf3b6071051 . Unfortunately the link won't provide a super easy way to generate this diagram. But it does provide an easy way to generate diagrams of this kind.

Reply:
That approach does not seem very maintainable. As stated, that's fine when we don't change the diagram.

An alternative like draw.io would allow us to host the "source" of the image in `.git`. Github would render the diagram when displaying a `.md` file. (I don't know what other `.md` renderers will do that, so the ascii diagram definitely has a leg up on portability.)

By [including an `edit-diagram.html`](https://desk.draw.io/support/solutions/articles/16000042371-embed-a-diagram-in-github-markdown) file in our repo, we can even directly edit the diagram, using the `draw.io` UI.